cpp/nacjd.cc

```diff
@@ -0,0 +1,361 @@
+/** \file    nacjd.cc
+    \brief   Identifies URLs that we can use for further processing.
+    \author  andreas-ub
+
+    \copyright 2021 Universitätsbibliothek Tübingen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <sstream>
+#include <unordered_set>
+#include <utility>
+#include <cerrno>
+#include <cstring>
+#include <getopt.h>
+#include "Downloader.h"
+#include "FileUtil.h"
+#include "HttpHeader.h"
+#include "JSON.h"
+#include "MARC.h"
+#include "UrlUtil.h"
+#include "util.h"
+#include "UBTools.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("marc_title_in_file new_marc_title_out_file\n"
+            "where marc_title_in_file contains also icpsr records (001 or 035a)\n"
+            "these records are not processed any more\n"
+            "new_marc_title_out_file contains all icpsr records not contained in input file.");
+}
+
+
+std::vector<std::string> ids_website;
+const unsigned int TIMEOUT_IN_SECONDS(15);
+const std::string NACJD_TITLES(UBTools::GetTuelibPath() + "nacjd_titles.html");
```

Comment:
If these are some temporary files, we should rather put them into "/tmp" or use FileUtil::AutoTempDirectory to store them.